src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-validators.ts

```diff
@@ -103,6 +103,17 @@ export class CdValidators {
     );
   }
 
+  /**
+   * Validator that performs SSL certificate validation of pem format.
+   * @returns {ValidatorFn} A validator function that returns an error map containing `pattern`
+   *   if the validation check fails, otherwise `null`.
+   */
+  static pemCert(): ValidatorFn {
+    return Validators.pattern(
+      /^-----BEGIN .+?-----(\n|\r|\f)((.+)?((\n|\r|\f).+)*)(\n|\r|\f)-----END .+?-----[\n\r\f]*$/
```

Comment:
You can heavily simplify this with [regex flags: `m` (multiline) and `s` (dotall)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#advanced_searching_with_flags):
```suggestion
      /^-----BEGIN .+-----$.+^-----END .+-----$/ms
```